doc/user/content/sql/create-source/kafka.md

```diff
@@ -104,11 +113,52 @@ echo ":" | kcat -b $BROKER -t $TOPIC -Z -K: \
   -X sasl.password=$KAFKA_PASSWORD
 ```
 
+#### Value decoding errors
+
+{{< private-preview />}}
+
+By default, if an error happens while decoding the value of a message for a
+specific key, Materialize sets the source into an error state. You can
+configure the source to continue ingesting data in the presence of value
+decoding errors using the `VALUE DECODING ERRORS = INLINE` option:
+
+```mzsql
+CREATE SOURCE kafka_upsert
+  FROM KAFKA CONNECTION kafka_connection (TOPIC 'events')
+  KEY FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_connection
+  VALUE FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_connection
+  ENVELOPE UPSERT (VALUE DECODING ERRORS = INLINE);
+```
+
+When this option is specified, the source will have the following schema,
+regardless of the specified `FORMAT`:
+
+Field         | Type                                                                                       | Nullable | Meaning
+--------------|--------------------------------------------------------------------------------------------|--------- |-----------
+`key`         | [[`record`](/sql/types/record/), [`bytea`](/sql/types/bytea/), [`text`](/sql/types/text/)] | `false`  | The Kafka message key. The type of this field depends on the specified `KEY FORMAT`. For unnamed formats (e.g., `TEXT`), the column name will be `key`. If the key is encoded using a format that includes schemas (e.g., `AVRO`), the column will take its name from the schema. The column can be renamed using the [`INCLUDE KEY AS...`](#key) syntax.
```

Comment:
Didn't test that the field indeed assumes whichever name is specified in the schema for a `key`, but did test that using `INCLUDE KEY AS` does the rename. Please double-check me on this, @rjobanp!

Reply:
Yes the key field naming semantics should be the same as before in this case